### PR TITLE
[File and Queue] Client constructors

### LINKED ISSUE #7806: [Storage-file] [Storage-queue] Update client constructors

### PR DESCRIPTION
**Pull request by rakshith91**
Resolves #7806 

**Breaking changes**

- Added new the following methods:
  -  `from_queue_url` in QueueClient
  -  `from_share_url` in ShareClient
  -  `from_directory_url` in DirectoryClient
   -  `from_file_url` in FileClient

- The default constructors for each of the above client now takes in only account_url with mandatory parameters such as share_name, file_path etc.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file/azure/storage/file/file_client.py [hunk 1]
```diff
@@ -167,22 +164,52 @@ def __init__( # type: ignore
...
         self._client = AzureFileStorage(version=VERSION, url=self.url, pipeline=self._pipeline)
 
+    @classmethod
+    def from_file_url(
+            cls, file_url,  # type: str
+            snapshot=None,  # type: Optional[Union[str, Dict[str, Any]]]
+            credential=None,  # type: Optional[Any]
+            **kwargs  # type: Any
+        ):
+        # type: (...) -> None
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/file_client.py]: Type hint should be returning `FileClient`

File: sdk/storage/azure-storage-file/azure/storage/file/share_client.py [hunk 2]
```diff
@@ -108,15 +106,47 @@ def __init__( # type: ignore
...
+    @classmethod
+    def from_share_url(cls, share_url,  # type: str
+            snapshot=None,  # type: Optional[Union[str, Dict[str, Any]]]
+            credential=None,  # type: Optional[Any]
+            **kwargs  # type: Any
+        ):
+        # type: (...) -> ShareClient
+        """
+        :param str share_url: The full URI to the share.
+        :param share: The share with which to interact. If specified, this value will override
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/share_client.py]: Doesn't look like we need `share` in the docstring.

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py [hunk 3]
```diff
@@ -59,12 +59,12 @@ class DirectoryClient(AsyncStorageAccountHostsMixin, DirectoryClientBase):
     :ivar str location_mode:
         The location mode that the client is currently using. By default
         this will be "primary". Options include "primary" and "secondary".
-    :param str directory_url:
-        The full URI to the directory. This can also be a URL to the storage account
-        or share, in which case the directory and/or share must also be specified.
-    :param share: The share for the directory. If specified, this value will override
+    :param str account_url:
+        The URI to the account. The method `from_directory_url` must be used in order to
```
Comment (johanste) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: This description needs some updates. The `must be used` sounds like it is imperative that something other than this method is used. It should be softened to something like "In order to create a client given the full URI to the directory, use the `from_directory_url` classmethod."

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py [hunk 4]
```diff
@@ -59,12 +59,12 @@ class DirectoryClient(AsyncStorageAccountHostsMixin, DirectoryClientBase):
...
+        use the full URI to the directory.
+    :param share_name: The share for the directory. If specified, this value will override
         a share value specified in the directory URL.
-    :type share: str or ~azure.storage.file.ShareProperties
+    :type share_name: str
     :param str directory_path:
```
Comment (johanste) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: Why does this still have a "if specified" clause in the documentation?

File: sdk/storage/azure-storage-file/azure/storage/file/directory_client.py [hunk 5]
```diff
@@ -114,17 +111,53 @@ def __init__( # type: ignore
...
+            The full URI to the directory.
+        :param str snapshot:
+            An optional share snapshot on which to operate.
+        :param credential:
+            The credential with which to authenticate. This is optional if the
+            account URL already has a SAS token. The value can be a SAS token string or an account
+            shared access key.
+        """
+        try:
+            if not directory_url.lower().startswith('http'):
```
Comment (johanste) [inline sdk/storage/azure-storage-file/azure/storage/file/directory_client.py]: The documentation says that this is a full URI, yet we prepend https:// ?

**Comment by rakshith91** [comment]
https://dev.azure.com/azure-sdk/public/_build/results?buildId=145126&view=results

Python - storage - ci is green....looks like there's  some issue with credentials for other pipelines ( not at all related to this change)
so merging it

**Pull request merged**